.github/workflows/process-codeql-sink-models.js

```diff
@@ -0,0 +1,135 @@
+        }
+
+        for (const dataEntry of data) {
+            if (dataEntry.length !== 5) {
+                throw new Error('Contains malformed data entry: ' + dataEntry)
+            }
+
+            // See https://github.com/github/codeql/blob/codeql-cli/v2.21.2/actions/ql/lib/codeql/actions/dataflow/internal/ExternalFlowExtensions.qll#L22-L24
+            /** @type string[] */
+            const [action, version, input, kind, provenance] = dataEntry
+            if (!relevantKinds.has(kind)) {
+                continue
+            }
+            if (onlyManualModels && provenance !== 'manual') {
+                continue
+            }
+
+            // Ignore 'on: workflow_call' inputs seem to be mostly for for 'generated' models
+            if (action.includes('/.github/workflows/')) {
+                continue
+            }
+
+            // Currently all models use only '*' as affected version, so for simplicity only
+            // support that for now
+            if (version !== '*') {
+                throw new Error(
+                    'Non-wildcard versions are not supported yet: ' + version
+                )
+            }
+
+            const inputPrefix = 'input.'
+            if (!input.startsWith(inputPrefix)) {
+                throw new Error('Contains input with unexpected format: ' + input)
+            }
+            const inputName = input.substring(inputPrefix.length)
+
+            let inputs = codeInjectionSinks.get(action)
+            if (inputs === undefined) {
+                inputs = []
+                codeInjectionSinks.set(action, inputs)
+            }
+            inputs.push(inputName)
+        }
+    }
+}
+
+/**
+ * @param {string} codeQlDir
+ * @param {fs.PathLike} outputFile
+ */
+function processModels(codeQlDir, outputFile) {
+    const modelsDir = path.join(codeQlDir, 'actions/ql/lib/ext')
+    const files = fs.readdirSync(modelsDir, {
+        recursive: true,
+        withFileTypes: true,
+    })
+
+    const relevantKinds = new Set(['code-injection'])
+    // For now only include models manually curated by the CodeQL developers
+    const onlyManualModels = true
+
+    let processedCount = 0
+    for (const file of files) {
+        if (file.isFile()) {
+            const name = file.name
+            if (name.endsWith('.yml') || name.endsWith('.yaml')) {
+                processedCount++
+                const filePath = path.join(file.parentPath, name)
+                try {
+                    processYamlFile(filePath, relevantKinds, onlyManualModels)
+                } catch (e) {
+                    throw new Error('Failed processing file: ' + filePath, { cause: e })
+                }
+            }
+        }
+    }
+
+    console.info(`Processed ${processedCount} files`)
+
+    // Important: Data format must match the parsing logic in the Rust code
+    const zizmorModelData = [...codeInjectionSinks].sort().map(entry => `${entry[0]}|${entry[1].join(',')}\n`).join('')
```

Comment:
Used `|` and `,` here are separators since they are probably not valid as part of repository or `inputs` name.

Reply:
Instead of creating our own syntax here, perhaps we could use JSON? That'll be slightly slower at initialization time in `zizmor` itself, but IMO it's worth it versus doing bespoke ser/de.